src/rules/utils.ts

```diff
@@ -633,6 +633,37 @@ export const isHook = (
   node.callee.type === AST_NODE_TYPES.Identifier &&
   HookName.hasOwnProperty(node.callee.name);
 
+export const isFunctionDeclarationTestCase = (
+  context: TSESLint.RuleContext<string, any>,
+  node: TSESTree.FunctionDeclaration,
+): boolean => {
+  return getFunctionDeclarationTestCallExpressions(context, node).length > 0;
+};
+
+export const getFunctionDeclarationTestCallExpressions = (
+  context: TSESLint.RuleContext<string, any>,
+  node: TSESTree.FunctionDeclaration,
+): Array<JestFunctionCallExpression<TestCaseName>> => {
+  const variables = context.getDeclaredVariables(node);
+  return variables.reduce<Array<JestFunctionCallExpression<TestCaseName>>>(
+    (acc, { references }) => {
+      const callExpressions = references
+        .map(({ identifier }) => identifier.parent)
+        .filter<TSESTree.CallExpression>(isCallExpression)
```

Comment:
It would be better to inline this guard, due to its simplicity & b/c it's generally not a good idea to have the guarded variable optional as it complicates the type narrowing.